### ISSUE TITLE
Is there a way to configure h2 client_header_timeout?

### ISSUE THREAD

**Issue by taikulawo** [hyperium/h2#823]
Hi,

We are add NGINX `client_header_timeout` support, Does h2 crate can configure read_timeout on per req level?

https://nginx.org/en/docs/http/ngx_http_core_module.html#client_header_timeout:~:text=Syntax%3A-,client_header_timeout,-time%3B
https://github.com/nginx/nginx/blob/43191be96fb5f425710400e0064c0f2d285a352d/src/http/v2/ngx_http_v2.c#L346

**Comment by seanmonstar** [comment]
No, there currently isn't an option to configure that. I wouldn't be against it's addition, but it'd likely need an implementation proposal first.

**Comment by taikulawo** [comment]
If we shouldn't include runtime timeout/sleep, how we implement timeout mechanism
